### DISCUSSION TITLE
How to automatically reload last checkpoint when loss unexpectedly spikes?

### DISCUSSION THREAD

**Discussion by RylanSchaeffer** [Lightning-AI/pytorch-lightning#14262] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I'm facing a problem where during training, my loss will unexpectedly spike, like so:

![image](https://user-images.githubusercontent.com/8942987/185260013-5e240d73-cfc2-44b2-aff3-919ccda604d8.png)

When this happens, I want to automatically reload the last checkpoint, reset the optimizer and resume training. How do I do this?
